packages/front-end/enterprise/components/Dashboards/DashboardEditor/index.tsx

```diff
@@ -312,59 +307,77 @@ function DashboardEditor({
         />
-        {isEditing && (
-          <Flex justify="center" mb="1em" position="relative">
-            {isDefined(i) && (hoverAddBlock === i || addBlockDropdown === i) && (
-              <div
-                style={{
-                  pointerEvents: "none",
-                  position: "absolute",
-                  top: "0",
-                  width: "100%",
-                  height: "9px",
-                  borderBottom: "1px solid var(--violet-a9)",
-                  zIndex: -1,
-                }}
-              />
-            )}
-            <AddBlockDropdown
-              onDropdownOpen={() => setAddBlockDropdown(i)}
-              onDropdownClose={() => setAddBlockDropdown(undefined)}
-              trigger={
-                <IconButton
-                  onMouseEnter={() => {
-                    setHoverAddBlock(i);
-                  }}
-                  onMouseLeave={() => {
-                    setHoverAddBlock(undefined);
+        <Container
+          py="1em"
+          onMouseEnter={() => {
+            if (!isDefined(addBlockDropdown)) setShowAddBlock(i);
+          }}
+          onMouseLeave={() => {
+            if (!isDefined(addBlockDropdown)) setShowAddBlock(undefined);
+          }}
+          className={clsx({
+            "dashboard-disabled": editDrawerOpen,
+          })}
+        >
+          {isEditing && (
+            <Flex justify="center" position="relative">
+              {isDefined(i) && (hoverAddBlock === i || addBlockDropdown === i) && (
+                <div
+                  style={{
+                    pointerEvents: "none",
+                    position: "absolute",
+                    top: "0",
+                    width: "100%",
+                    height: "50%",
+                    borderBottom: "1px solid var(--violet-a9)",
+                    zIndex: -1,
                   }}
-                  className={clsx({
-                    "dashboard-disabled": editDrawerOpen,
-                  })}
-                  size="1"
-                >
-                  <Tooltip
-                    body="Add block"
-                    tipPosition="top"
-                    delay={0}
-                    state={hoverAddBlock === i && addBlockDropdown !== i}
-                    ignoreMouseEvents
-                    innerClassName="px-0 py-1"
+                />
+              )}
+              <AddBlockDropdown
+                onDropdownOpen={() => setAddBlockDropdown(i)}
+                onDropdownClose={() => {
+                  setAddBlockDropdown(undefined);
+                  setShowAddBlock(undefined);
+                }}
+                trigger={
+                  <IconButton
+                    onMouseEnter={() => {
+                      setHoverAddBlock(i);
+                    }}
+                    onMouseLeave={() => {
+                      setHoverAddBlock(undefined);
+                    }}
+                    className={clsx({
+                      "d-none":
```

Comment:
the layout shift here is a bit jarring. could we instead toggle visibility and pointerEvents?